python/spark_package/spark_package.py

```diff
@@ -365,16 +367,17 @@ def publish_release(name, user, token, folder, version, out, zip):
         zip = zip_artifact(folder, name, version, out)
     binary_zip = ""
     if zip is not None:
-        with open(zip) as f:
+        with open(zip, 'rb') as f:
```

Comment:
I remember this causing some "malformed jar" exception when running with Spark after using `--packages` to import the package. Did you get a chance to try it end-to-end?